deephyper/search/hps/_cbo.py

```diff
@@ -524,20 +527,110 @@ def _return_forbid(self, cond, cst_new):
                 print("Not supported type" + str(type(cond)))
         return cond_new
 
-    def fit_surrogate(self, df):
+    def clean_dataframe(self, df):
+        # Ensure the DataFrame is of correct type
+        if not isinstance(df, pd.DataFrame):
+            raise ValueError("Provided object is not a pandas DataFrame")
+
+        # Check if objective columns exist
+        if "objective" not in df.columns and not any(
```

Comment:
Since python 3.8, this could be simplified as 
```python
if not (objective_columns := df.filter(regex=r"^objective(?:_\d+)?$").columns):
    raise ValueError(...)
```